OpenRA.Game/Map/Map.cs

```diff
@@ -285,6 +287,26 @@ public static string ComputeUID(IReadOnlyPackage package)
 			}
 		}
 
+		// PERF This is a way to get MapFormat without expensive yaml parsing
+		public static int GetMapFormat(IReadOnlyPackage p)
+		{
+			var result = "";
+			foreach (var line in p.GetStream("map.yaml").ReadAllLines())
+			{
+				var search = Regex.Match(line, "MapFormat: (.?\\d)");
+				if (search.Success && search.Groups.Count > 0)
+				{
+					result = search.Groups[1].Value;
+					break;
+				}
+			}
+
+			if (result == "")
+				throw new InvalidDataException($"Does not contain MapFormat");
```

Comment:
Change to `throw new InvalidDataException($"MapFormat is not definedt\n File: {p.Name}");` to match the exception below.

Reply:
ok